File: lab_02/src/main.py
import time
import sys

def compute(x: float) -> float:
    return x*x - x*x + x*4 - x*5 + x + x

def main():
    while True:
        user_input = input("Enter number of iterations (or non-number to exit): ")
        try:
            n = int(user_input)
        except ValueError:
            print("Invalid input. Exiting.")
            break

        x = 1.0
        start = time.perf_counter()

        for _ in range(n):
            res = compute(x)

        end = time.perf_counter()
        print(f"Time taken: {end - start:.6f} seconds")

        again = input("Run again? (y/n): ").strip().lower()
        if again != 'y':
            break

if __name__ == "__main__":
    main()

File: lab_02/src/main.cpp
#include <iostream>
#include <ctime>

double compute(double x) {
    return x*x - x*x + x*4 - x*5 + x + x;
}

int main() {
    while(true) {
        long n; 
        std::cout << "Enter number of iterations (or non-number to exit): ";

        if (!(std::cin >> n)) {
            std::cout << "Invalid input. Exiting.\n";
            break;
        }

        double x;
        clock_t start = clock();

        for (long long i = 0; i < n; ++i) {
            volatile double res = compute(x); // volatile чтобы не оптимизировалось
        }

        clock_t end = clock();

        double seconds = (double)(end - start) / CLOCKS_PER_SEC;
        
        std::cout << "Time taken: " << seconds << " seconds\n";

        
        char again;
        std::cout << "Run again? (y/n): ";
        std::cin >> again;
        if (again != 'y' && again != 'Y') {
            break;
        }
    }
    return 0;
}
